app/assets/javascripts/modules/mobile-filters-modal.js

```diff
@@ -0,0 +1,139 @@
+    this.$module = $module[0]
+    this.$facetsBox = this.$module.querySelector('.facets__box')
+    this.$closeTriggers = this.$module.querySelectorAll('.js-close-filters')
+    this.$clearFiltersTrigger = this.$module.querySelector('.js-clear-selected-filters')
+    this.$body = document.querySelector('body')
+
+    this.$module.open = this.handleOpen.bind(this)
+    this.$module.close = this.handleClose.bind(this)
+    this.$module.clearFilters = this.handleClearFilters.bind(this)
+    this.$module.ModalFocus = this.handleModalFocus.bind(this)
+    this.$module.boundKeyDown = this.handleKeyDown.bind(this)
+
+    var $triggerElement = document.querySelector(
+      '[data-toggle="mobile-filters-modal"][data-target="' + this.$module.id + '"]'
+    )
+    if ($triggerElement) {
+      $triggerElement.addEventListener('click', this.$module.open)
+    }
+
+    if (this.$closeTriggers) {
+      for (var t = 0; t < this.$closeTriggers.length; t++) {
+        this.$closeTriggers[t].addEventListener('click', this.$module.close)
+      }
+    }
+
+    if (this.$clearFiltersTrigger) {
+      this.$clearFiltersTrigger.addEventListener('click', this.$module.clearFilters)
+    }
+  }
+
+  MobileFiltersModal.prototype.handleOpen = function (event) {
+    if (event) {
+      event.preventDefault()
+    }
+    // when our support browser matrix is above iOS 13
+    // we can use overflow: hidden on the <body> and remove position: fixed
+    this.$body.style.top = '-' + window.scrollY + 'px'
+    this.$body.style.position = 'fixed'
+    this.$focusedElementBeforeOpen = document.activeElement
+    this.$module.style.display = 'block'
+    this.$facetsBox.setAttribute('aria-modal', true)
+    this.$facetsBox.setAttribute('tabindex', 0)
+    this.$facetsBox.focus()
+
+    document.addEventListener('keydown', this.$module.boundKeyDown, true)
+  }
+
+  MobileFiltersModal.prototype.handleClose = function (event) {
+    if (event) {
+      event.preventDefault()
+    }
+
+    var offsetTop = this.$body.style.top
+    this.$body.style.position = ''
+    this.$body.style.top = ''
+    window.scrollTo(0, parseInt(offsetTop || '0') * -1)
+    this.$module.style.display = 'none'
+    this.$facetsBox.removeAttribute('aria-modal')
+    this.$facetsBox.removeAttribute('tabindex')
+    this.$focusedElementBeforeOpen.focus()
+
+    document.removeEventListener('keydown', this.$module.boundKeyDown, true)
+  }
+
+  MobileFiltersModal.prototype.handleModalFocus = function () {
+    this.$facetsBox.focus()
+  }
+
+  // while open, prevent tabbing to outside the dialogue
+  // and listen for ESC key to close the dialogue
+  MobileFiltersModal.prototype.handleKeyDown = function (event) {
+    var KEY_TAB = 9
+    var KEY_ESC = 27
+
+    switch (event.keyCode) {
+      case KEY_TAB:
+        if (event.shiftKey) {
+          if (document.activeElement === this.$facetsBox) {
+            event.preventDefault()
+            this.$closeTrigger.focus()
```

Comment:
The 'focus trap' doesn't work as expected (tested on [https://finder-frontend-pr-1804.herokuapp.com/search/all](https://finder-frontend-pr-1804.herokuapp.com/search/all)) as `this.$closeTrigger` is not a thing (we now have a set of `this.$closeTriggers` instead of one). We can select the last one or the one in `facets__footer` if we'll always have that button as the last interactive element judging by the markup.

Reply:
🤦‍♂ good spot @alex-ju 
Decided to focus on the first instance of the trigger

Reply:
Hmm.. still doesn't seem to be working for me on the Heroku preview – after I tab the 'Show x results' button the next focused element is outside the modal. Happy to have a second look at the code whenever you think it's ready.

Reply:
oh that! 
I think i fixed that now 🤞